### ISSUE TITLE
How to use default config values if not present in environment

### ISSUE THREAD

**Issue by haneefkassam** [FoalTS/foal#1071]
I'm trying to set up configuration (while also using the environment files) so that there is a fallback.  For example, I have the following configs (I've trimmed them down for simplicity):

*default.json*
```json
{
  "database": {
    "username": "user",
    "password": "pass"
  }
}
```
```javascript
// production.js
const { Env } = require('@foal/core');

module.exports = {
  "database": {
    "username": Env.get("USERNAME") || <want-to-use-default-here>,
    "password": Env.get("PASSWORD") || <want-to-use-default-here>
  }
}
```

In the above situation, I'm looking for the credentials in the environment (either in the `.env` file or the nodejs process environment).  If the value returned is undefined, I'd like to use the value from the default.json config.  However, I don't see, in code, any way I can use the `Config.get` function to do this, and because a value is returned from `Env.get` (even if that value is `undefined`), it sets the property, so doesn't fallback to the one defined in default on its own (in the case that the property doesn't exist).

Is there a way to do this simply?  or do I need to add the defaults to each of the environment config files (thus making `default.json` pointless).

**Comment by LoicPoullain** [comment]
Hi @haneefkassam  👋 

The reason behind this behavior is that, under the hood, Foal makes a deep `assign` of the config objects of each environment. This is why this does not work with `undefined` values.

I'll fix this in v3 (as it introduces a breaking change).

In the meantime, the only thing that I can see here is to use the solution that you suggested here `|| myVar`.

**Comment by LoicPoullain** [comment]
Behavior fixed in v3.0.0
